### PR TITLE
[AOTI] Add more ABI-compatiblity unit test

### PR DESCRIPTION
**Pull request by desertfire**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #123900

Summary: Follow https://github.com/pytorch/pytorch/pull/123848, and test more c10 util functions.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @chauhang

### CODE REVIEW

File: torch/_inductor/codegen/cpp_prefix.h [hunk 1]
```diff
@@ -15,7 +15,6 @@
 
 #include <ATen/NumericUtils.h>
 #include <ATen/core/PhiloxRNGEngine.h>
-#include <ATen/native/Math.h>
```
Comment (chenyang78) [inline torch/_inductor/codegen/cpp_prefix.h:18]: Just double-check if this change was intentional.

**Review by chenyang78 — APPROVED** [review]
Left a minor comment. LGTM otherwise. Thanks!

File: torch/_inductor/codegen/cpp_prefix.h (hunk 1 above)
Comment (desertfire) [reply torch/_inductor/codegen/cpp_prefix.h:18]: Yes, it is. I tried to reduce the headers included here, and this one doesn't seem needed anymore.

**Pull request closed without merging**